File: python/bob/bob.py
def hey(phrase):
    if question(phrase)       : return 'Sure.'
    if shouting(phrase)       : return 'Whoa, chill out!'
    if starts_with_whitespace : return 'Whatever.'
    if ends_with_whitespace   : return 'Sure.'
    if silence                : return 'Fine. Be that way!'
    if prolonged_silence      : return 'Fine. Be that way!'
    else: return 'Whatever.'

def question(phrase):
    if phrase[-1] == '?' : return True

def shouting(phrase):
    if phrase == phrase.upper() : return True

def starts_with_whitespace(phrase):
    if phrase[0] == ' ' : return True

def ends_with_whitespace(phrase):
    if phrase[-1] == ' ' : return True

def silence(phrase):
    if phrase == '' : return True

def prolonged_silence(phrase):
    if phrase == '    ' : return True

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if (input === input.toUpperCase()) {
    return "Whoa, chill out!"
  } else if (input.charAt(input.length - 1) == "?"){
    return "Sure."
  } else {
    return "Whatever."
  }
};

module.exports = Bob;
